doc: Delta Notifications for Resource Changes

dirkwa opened the pull request:
As discussed in discord channel #specification the documentation is lagging the emt of deltas to minimize server reloads and better user expirience.

panaaj commented on docs/develop/plugins/resource_provider_plugins.md:
```diff
@@ -176,3 +176,332 @@ module.exports = function (app) {
+1. **Upload** - New resource added
+2. **Delete** - Resource removed
+3. **Modify** - Resource updated (rename, move, etc.)
+4. **Toggle** - Resource enabled/disabled
```
Not sure I understand "toggle" in this in the client context.
What delta message would this action emit? I would expect to the client it would just appear as resource updated or deleted deltas.

dirkwa commented:
Thank you for the thorough review! You've raised excellent points that improve both the technical accuracy and clarity of the documentation:

**1. Signal K v2 for resource deltas**
You're absolutely right - I've updated all examples to use `2` directly instead of the `skVersion` variable, since resources shouldn't be in the full model cache. Added clarifying comments to explain this.

**2. Simplified action types**
Good catch. "Toggle" is indeed just an update operation from the delta perspective. I've simplified the action types to **Create**, **Update**, and **Delete**, with enable/disable now shown as an example of Update operations. This is clearer and more aligned with standard CRUD patterns.

**3. "Upload" → "Create"**
Agreed - "Upload" was too specific and implied only HTTP uploads. I've changed the terminology to "Create" throughout, with examples clarifying that resources can be added via upload, file copy, download, scan, etc. This makes the documentation more broadly applicable.

panaaj commented:
Looking further at this... it appears that the intent is to address the the need for a provider to emit notifications when resources are added, updated or deleted out-of-band _(e.g. using the provider UI or adding / deleting files from a filesystem)_ not via the Resource API endpoints. 

I think that this is a topic that should be added to the documentation but feel the topic heading and language should be more direct to reflect this.

IMHO there is quite a bit which could be omitted without losing the message (this may because some content may have been AI generated).

tkurki commented:
Yes, this is exactly what I was hoping for: that our central documentation improves over time, omissions are addressed  and we accumulate knowledge that contributors can benefit from and make the system more useful.

Instead of commenting here I made a PR to your PR with a bunch of suggested changes. Mostly it is about making the documentation **more concise** (per my original comment in Slack) and read more like neutral documentation instead of trying to promote something.

dirkwa commented:
@tkurki Sorry, missed your PR, all merged. Thanks for your support.
